### PR TITLE
fix: split stream query result to avoid grpc response too large error

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/36089

### CODE REVIEW

File: internal/util/streamrpc/streamer.go [hunk 1]
```diff
@@ -102,25 +102,66 @@ func mergeCostAggregation(a *internalpb.CostAggregation, b *internalpb.CostAggre
 	return &internalpb.CostAggregation{
 		ResponseTime:         a.GetResponseTime() + b.GetResponseTime(),
 		ServiceTime:          a.GetServiceTime() + b.GetServiceTime(),
-		TotalNQ:              a.GetTotalNQ() + b.GetTotalNQ(),
+		TotalNQ:              a.GetTotalNQ(),
```
Comment (tedxu) [inline internal/util/streamrpc/streamer.go:105]: Why is this changed?

File: internal/util/streamrpc/streamer.go [hunk 2]
```diff
@@ -133,11 +174,18 @@ func (s *ResultCacheServer) Send(result *internalpb.RetrieveResults) error {
 	}
 
 	s.cache.Put(result)
-	if s.cache.IsFull() {
+	if s.cache.IsFull() && s.cache.size <= s.maxMsgSize {
 		result := s.cache.Flush()
 		if err := s.srv.Send(result); err != nil {
 			return err
 		}
+	} else if s.cache.IsFull() && s.cache.size > s.maxMsgSize {
```
Comment (tedxu) [inline internal/util/streamrpc/streamer.go:182]: Is it possible that we return the results capped with `maxMsgSize` in `s.cache.Flush()`, instead of splitting messages and send them separately?

We will introduce 2 problems by splitting messages: 1) the additional memory copy will harm the total performance, and 2) we'll send a small result, which is the last element of the return of the split, every time when cache is full.

File: pkg/util/paramtable/component_param.go [hunk 3]
```diff
@@ -3094,11 +3095,20 @@ user-task-polling:
 		Key:          "queryNode.queryStreamBatchSize",
 		Version:      "2.4.1",
 		DefaultValue: "4194304",
-		Doc:          "return batch size of stream query",
+		Doc:          "return min batch size of stream query",
```
Comment (tedxu) [inline pkg/util/paramtable/component_param.go:3098]: Can we validate if the min size is less than max? What is the consequence if they are configured wrong?

File: internal/util/streamrpc/streamer.go (hunk 1 above)
Comment (aoiasd) [reply internal/util/streamrpc/streamer.go:105]: The old TotalNq is wrong but not used before. Nq should be same between results in same query.

File: pkg/util/paramtable/component_param.go (hunk 3 above)
Comment (aoiasd) [reply pkg/util/paramtable/component_param.go:3098]: We will flush till buffer size large than min, then split message size less than max.
If max less than min, we will split all message.

File: internal/util/streamrpc/streamer.go (hunk 2 above)
Comment (aoiasd) [reply internal/util/streamrpc/streamer.go:182]: The split don't have large memory copy, it reference part of buffer result ids directly.
